### ISSUE TITLE
Make RefinementSingleSelect and some other links not a NextLink

### ISSUE THREAD

**Issue by sterlinghirsh** [iFixit/react-commerce#818]
## Problem
When loading https://www.ifixit.com/Parts/iPhone_12 it looks like we're prefetching things that shouldn't be prefetched because of using NextLink. These links are meant to be handled with Algolia, so we shouldn't be prefetching them. We still want to include the url in the href for SEO reasons.

![image](https://user-images.githubusercontent.com/589425/195424986-547818ea-99f8-4abc-8ad8-23e219624203.png)

I've also noticed that some links in the header are attempting to prefetch.

![image](https://user-images.githubusercontent.com/589425/195427213-4e56c33b-d6fb-4e69-afc7-cabd140089df.png)

## Do This
Make these normal links so they don't waste a bunch of round trips prefetching.

Note: It's possible that this will fix some bugs we've had with algolia's history control conflicting with next's history control, so we might be able to get rid of the hack we have to fix the back button.

**Comment by dhmacs** [comment]
We should be careful about changing url without Next being aware of it.. it will likely cause the page props to go out of sync with the page url (e.g. page title, description, sections, anything that is fetched server side). 
I'd say that unless we're only changing search params, it's a good idea to let Next.js manage url.

**Comment by danielbeardsley** [comment]
> I'd say that unless we're only changing search params, it's a good idea to let Next.js manage url.

I'm not sure here, but I think we currently intercept the clicks on these links and algolia updates the url.

Also, the point of this issue is that we shouldn't be pre-fetching 10 other next.js pages that aren't actually needed. Note, he also mentioned that non-next.js pages are also being pre-fetched.

**Comment by sterlinghirsh** [comment]
I am in favor of being careful, but it's pretty easy to try it out and then see if it breaks. It seems like it should work to let Algolia change the url and then have next resume control when we switch. But also if we link to 10 products from a collection using NextLink, is it going to preload all those? Seems like it could be a lot of stuff to load, but I guess we'll have to experiment. I don't know how practical it is, but if we run into a conflict with Algolia and Next, maybe it's possible that the first time on a page Algolia starts changing the url, we opt out of the NextLink optimization and handle any links as full pageloads. I'd rather not if it's not necessary, but let's experiment with it.

**Comment by dhmacs** [comment]
@sterlinghirsh 

> When loading https://www.ifixit.com/Parts/iPhone_12 it looks like we're prefetching things that shouldn't be prefetched because of using NextLink. These links are meant to be handled with Algolia, so we shouldn't be prefetching them.

Now that I double checked this, I have two questions. 
1. Is this a screenshot from the network tab? Where do you see that's prefetching all these things? When I go to that page and I check the network tab I don't see any of these 🤔 
2. Why do you think Next Link is responsible for that? I see these are json files, but Next.js link prefetching only downloads the js for the page, not the data (unless we're using static generation, and we aren't since we're doing SSR)

>  if we link to 10 products from a collection using NextLink, is it going to preload all those?

Not the product data, just the code of the page, that means a single js file for the product page


@danielbeardsley 

> I'm not sure here, but I think we currently intercept the clicks on these links and Algolia updates the url.

Yes we do, and it has problems. Check this page https://www.ifixit.com/Parts/iPhone/Adhesive_Pads .
When it loads the title flicker because the logic runs client side. If you disable js to simulate how the google would see the page during the first pass, you see it renders the parent title instead of the item type.

**Comment by danielbeardsley** [comment]
> Where do you see that's prefetching all these things? When I go to that page and I check the network tab I don't see any of these

Same here, no sign of them. Though I wonder if this is because we just deployed #826 ? The pull that added the middleware was merged the about the same time we saw a 5x jump in requests.

> Why do you think Next Link is responsible for that? I see these are json files, but Next.js link prefetching only downloads the js for the page, not the data 

Huh.. that's a good point. *mystery deepens*

**Comment by sterlinghirsh** [comment]
Yes it was from the network tab, but looking again, I don't see it prefetching all these things anymore.

**Comment by dhmacs** [comment]
So if we can't reproduce the behaviour anymore, can we close this?

**Comment by danielbeardsley** [comment]
> Why do you think Next Link is responsible for that? I see these are json files, but Next.js link prefetching only downloads the js for the page, not the data (unless we're using static generation, and we aren't since we're doing SSR)

I noticed these requests are the same ones you get when you click the filters. Almost as if all the links were "clicked" on page load. Any chance you had some browser plugin that was auto-clicking all links or something?

**Comment by danielbeardsley** [comment]
Nope. It was the middleware. Confirmed.

I made a new branch that re-added the middleware and tested it:
https://react-commerce-prod-git-middleware-test-it-again-ifixit.vercel.app/

![image](https://user-images.githubusercontent.com/26855/195896292-f044940b-7073-4e9b-b73c-7efd0a123b42.png)

This is crazy that adding this small middleware would end up causing the page to request *every* next link on page load. This sounds like the kind of thing we need to open a vercel issue for.

https://github.com/iFixit/react-commerce/compare/middleware-test-it-again
